terragrunt/aws/cloudfront/cloudfront.tf

```diff
@@ -0,0 +1,55 @@
+resource "aws_cloudfront_origin_access_identity" "origin_access_identity" {
+  comment = "cloudfront origin access identity"
+}
+
+resource "aws_cloudfront_distribution" "distribution" {
+  for_each            = var.s3_bucket_regional_domain_name
+  default_root_object = "index.html"
+  enabled             = true
+  web_acl_id          = aws_wafv2_web_acl.cds_website_waf.arn
+
+  default_cache_behavior {
+    target_origin_id       = each.value
+    viewer_protocol_policy = "redirect-to-https"
+    allowed_methods        = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods = [
+      "GET",
+      "HEAD",
+    ]
+    forwarded_values {
+      query_string = false
+
+      cookies {
+        forward = "all"
+      }
+    }
```

Comment:
Could you try using the AWS managed cache and header policies - here's an example of what that looks like:
https://github.com/cds-snc/gcds-terraform/blob/1ffb8d00f6b34da5d9d8eec9f262d1abbe96f56b/terragrunt/aws/cdn/cloudfront.tf#L24-L29

Could you also add in the following:
```terraform
    viewer_protocol_policy = "redirect-to-https"
    compress               = true
```